python/pfs/drp/stella/readLineList.py

```diff
@@ -46,35 +48,43 @@ def run(self, detectorMap=None, metadata=None):
             Lines from the linelist.
         """
         lines = ReferenceLineSet.empty()
-        for filename in self.config.lineListFiles:
-            lines.extend(ReferenceLineSet.fromLineList(filename))
+        if self.config.lineListFiles:
+            for filename in self.config.lineListFiles:
+                lines.extend(ReferenceLineSet.fromLineList(filename))
         if self.config.restrictByLamps:
-            lamps = self.getLamps(metadata)
-            lines = self.filterByLamps(lines, lamps)
+            lamps, lampElementList = self.getLampInfo(metadata)
+            if lamps:
+                if not self.config.lineListFiles:
+                    for lamp in lamps:
+                        lines.extend(ReferenceLineSet.fromLineList(f'{lamp}.txt'))
```

Comment:
This file naming scheme needs to be documented somewhere.
I'd prefer a scheme where there's a configuration parameter mapping from lamp name to filename, and I can specify a list of lamps (not files) to activate.